### ISSUE TITLE
Cannot share files using public link after 7.0.3 update from 7.0.2

### ISSUE THREAD

**Issue by anaqreon** [owncloud/core#12320]
### Steps to reproduce (all in web interface)
1. Create folder
2. Share the folder using a public link, allowing public upload
3. Add files to the folder, then try to view them after upload "completes"

### Expected behaviour
The uploaded files should be available for download

### Actual behaviour
After upload, a yellow banner appears with message "Internal server error". The uploaded files are listed, but there is no content (no thumbnail and cannot preview or download)

### Server configuration
**Operating system**:
Debian 3.2.41-2+deb7u2 i686 GNU/Linux
**Web server:**
nginx version: nginx/1.4.4
**Database:**
MySQL
**PHP version:**
PHP 5.4.35-0+deb7u2 (cli) (built: Nov 19 2014 09:05:45) 
**ownCloud version:** (see ownCloud admin page)
7.0.3
**Updated from an older ownCloud or fresh install:**
Updated from 7.0.2 (after which the problem started)
**List of activated apps:**
Maps
**The content of config/config.php:**

```
<?php
$CONFIG = array (
  'datadirectory' => '/var/lib/owncloud/',
  'dbtype' => 'mysql',
  'version' => '7.0.3.4',
  'installedat' => '1331517878.793',
  'lastupdatedat' => '1337132624.292',
  'installed' => true,
  'versions' => 'true',
  'instanceid' => '507a11b862b93',
  'loglevel' => '1',
  'theme' => '',
  'maintenance' => false,
  'appstoreenabled' => false,
  'check_for_working_htaccess' => false,
  'check_for_working_webdav' => false,
  'passwordsalt' => '',
  'maxZipInputSize' => 838860800,
  'allowZipDownload' => true,
  'trusted_domains' => 
  array (
    0 => '',
  ),
  'dbname' => '',
  'dbhost' => 'localhost',
  'dbuser' => '',
  'dbpassword' => '',
);
```

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

### Client configuration
**Browser:**
FireFox, Chrome
**Operating system:**
MacOS
#### Web server error log
nothing
#### ownCloud log (data/owncloud.log)
```
{"app":"PHP","message":"Call to a member function getUID() on a non-object at \/var\/www\/my_webapp\/apps\/files_external\/lib\/config.php#124","level":3,"time":"2014-11-20T11:15:07+00:00"}
{"app":"PHP","message":"Call to a member function getUID() on a non-object at \/var\/www\/my_webapp\/apps\/files_external\/lib\/config.php#124","level":3,"time":"2014-11-20T11:15:08+00:00"}
```

**Comment by PVince81** [comment]
Did you guys have any more luck with 7.0.4 ?

**Comment by anaqreon** [comment]
I actually migrated to a different server that is still running 7.0.2, so I don't have the problem now.

**Comment by PVince81** [comment]
This should be fixed post 7.0.4. But in some cases broken share entries need to be reset by unsharing and resharing, nothing that can be done automatically. The bogus situation should not happen any more.
